src/components/GithubLogin.tsx

```diff
@@ -0,0 +1,135 @@
+/**
+ * You need to signin via credentials, If not then it will logout
+ * Inputbox to write github username (to link with existing userid)
+ * -> Prompt to githublogin
+ * -> Check DB for linked username
+ * -> If fount link the account else show error to try again
+ */
+
+'use client';
+import axios from 'axios';
+import { signIn, signOut, useSession } from 'next-auth/react';
+import { useRouter } from 'next/navigation';
+import { useEffect, useState } from 'react';
+
+export default function GithubLogin({
+  userId,
+  session,
+}: {
+  session: any;
+  userId: string;
+}) {
+  const router = useRouter();
+  const { status } = useSession();
+  const [value, setValue] = useState<string>('');
+  const [error, setError] = useState<string>('');
+  const [isClicked, setIsClicked] = useState<boolean>(false);
```

Comment:
instead of isClicked you can use IsPending or isLoading for this, it will better naming

Reply:
used isLoading, thanks for it